fe/fe-core/src/main/java/com/starrocks/sql/analyzer/BackupRestoreAnalyzer.java

```diff
@@ -84,16 +89,40 @@ public Void visitBackupStatement(BackupStmt backupStmt, ConnectContext context)
                     tableRefs.add(tableRef);
                 }
             }
+
+            Map<Long, TableRef> tableIdToTableRefMap = Maps.newHashMap();
+            Map<String, MaterializedView> mvNameMVMap = Maps.newHashMap();
             for (TableRef tableRef : tableRefs) {
-                analyzeTableRef(tableRef, dbName, database, tblPartsMap, context.getCurrentCatalog());
+                analyzeTableRef(tableRef, dbName, database, tblPartsMap, context.getCurrentCatalog(),
+                        mvNameMVMap, tableIdToTableRefMap);
                 if (tableRef.hasExplicitAlias()) {
                     throw new SemanticException("Can not set alias for table in Backup Stmt: " + tableRef,
                             tableRef.getPos());
                 }
             }
 
-            tableRefs.clear();
-            tableRefs.addAll(tblPartsMap.values());
+            if (!mvNameMVMap.isEmpty()) {
+                // check base table existed in the same db.
+                for (Map.Entry<String, MaterializedView> e : mvNameMVMap.entrySet()) {
+                    MaterializedView mv = e.getValue();
+                    for (BaseTableInfo baseTableInfo : mv.getBaseTableInfos()) {
+                        if (!tableIdToTableRefMap.containsKey(baseTableInfo.getTableId())) {
```

Comment:
base table and mv should be in the same database? why?

Reply:
For creating, mv's db and table's db may not be the same.

however, backup/restore only supports for the same database or only table. When we backup/restore the mv which is different db from the base table,  we can only backup/restore the data, and cannot ensure the mv is active at the final.